### PR TITLE
feat(workspaces): default workspace project role

### PR DESCRIPTION
**Pull request by cdriesler**
## Description & motivation

- Workspaces currently default all new members to `stream:reviewer`. This was done out of an abundance of caution, but doesn't quite match most users' expectations.
- We've opted to enable this "default workspace project role" as a setting on workspaces

## Changes:

- Adds `defaultProjectRole` to workspaces
- Uses this during `getWorkspaceRoleToDefaultProjectRoleMapping`
- A number of small changes in orbit of this:
  - Collect `updateWorkspace` input sanitization into a few helpers
  -  `PartialNullable<T>` utility type
  - Assorted test changes

### CODE REVIEW

File: packages/server/modules/workspaces/domain/operations.ts [hunk 1]
```diff
@@ -187,14 +187,11 @@ export type GetUserIdsWithRoleInWorkspace = (
...
-    name?: string | null
-    description?: string | null
-    logo?: string | null
-    defaultLogoIndex?: number | null
-    discoverabilityEnabled?: boolean | null
-    domainBasedMembershipProtectionEnabled?: boolean | null
-  }
+  workspaceInput: PartialNullable<
+    Omit<Workspace, 'id' | 'createdAt' | 'updatedAt' | 'defaultProjectRole'> & {
+      defaultProjectRole: string
```
Comment (gjedlicska) [inline packages/server/modules/workspaces/domain/operations.ts:192]: there is a type for project roles

File: packages/server/assets/workspacesCore/typedefs/workspaces.graphql [hunk 2]
```diff
@@ -259,6 +263,10 @@ type Workspace {
   """
   defaultLogoIndex: Int!
   """
+  The default role workspace members will receive for workspace projects.
+  """
+  defaultProjectRole: String
```
Comment (gjedlicska) [inline packages/server/assets/workspacesCore/typedefs/workspaces.graphql:268]: This should be String!

File: packages/server/modules/workspacesCore/domain/types.ts [hunk 3]
```diff
@@ -27,6 +27,7 @@ export type Workspace = {
   updatedAt: Date
   logo: string | null
   defaultLogoIndex: number
+  defaultProjectRole: Exclude<StreamRoles, 'stream:owner'>
```
Comment (gjedlicska) [inline packages/server/modules/workspacesCore/domain/types.ts:30]: expose this as a type from here

**Review by gjedlicska — CHANGES_REQUESTED** [review]
smol changes noted

**Review by cdriesler — COMMENTED** [review]
testing

**Review by gjedlicska — APPROVED** [review]

**Comment by cdriesler** [comment]
Ready for re-review, some additional changes after we spoke because the validation was also required in test setup

**Review by gjedlicska — APPROVED** [review]

**Pull request merged**